### ISSUE TITLE
When having nested cd, SC2013 is not raised

### ISSUE THREAD

**Issue by zapashcanon** [koalaman/shellcheck#1146]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2013
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

set -e

( cd blah
  ls
  cd -
  
  cd bloh
  ls

)

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Should raise SC2013 asking to write something like:

```sh

#!/bin/bash

set -e

( ( cd blah
  ls
  )
  
  cd bloh
  ls

)

```

**Comment by Dmole** [comment]
That could triple the size of a script by requiring a cd and shell for every single command.

**Comment by matthewpersico** [comment]
There's already a lot of cds in that script. I think the OP's point was that when you find `cd -`, you start a subshell-creating pair of () before the latest cd prior to the `cd -` and ending after the `cd -`, and then remove the `cd -`.

**Comment by Dmole** [comment]
But with `set -e` there is no advantage to sub shells, like all 4 of the following should be valid;

```
#!/bin/bash

set -e
cd blah
ls
cd bloh
ls
```

```
#!/bin/bash

set -e
cd blah
ls
cd ../bloh
ls
```

```
#!/bin/bash

(cd blah
ls
(cd bloh
ls))
```

```
#!/bin/bash

(cd blah
ls)
(cd bloh
ls)
```

**Comment by matthewpersico** [comment]
But none of those contain `cd -`. The OP's point was that 

```
cd foo
##... do stuff
cd -
```

should be warned with suggested replacement

```
( cd foo
## ...do stuff
)
```

**Comment by Dmole** [comment]
Thanks for being persistent, I totally missed that, good thing we have shellcheck.... but it's more the 2 consecutive cd-s in a row, and even then what if the script were not run from ~/ in the first place, then the OPs recommended solution would have a different result than the OPs sample.

only if it were 

```
#!/bin/bash

set -e
cd -
cd blah
ls
cd -
cd bloh
ls
```

would the suggestion be equivalent
#!/bin/bash

```
set -e
(
cd -
cd blah
ls
)
cd bloh
ls

```

**Comment by matthewpersico** [comment]
cd and cd~/ are synonyms. cd - means "cd to the directory you were in before the last cd".

```
$ cd

$ pwd
/home/mpersico5

$ cd /tmp

$ pwd
/tmp

$ cd /usr

$ pwd
/usr

$ cd -
/tmp

$ pwd
/tmp
```

**Comment by Dmole** [comment]
Thanks for being persistent, so (`cd -` == `cd $OLDPWD`) and (`cd` == `cd ~/` ), guess it's just one of those basic things I somehow missed or was never exposed to.
I removed some of my past comments to avoid derailing this issue.

**Comment by matthewpersico** [comment]
Nah, that's a bad thing. Now my comments are out of place...

**Comment by matthewpersico** [comment]
The example given is overly complicated. The simplest example would be

```
#!/bin/bash
cd blah
ls
cd -
```
which the OP wants to trigger:
````
WARNING: cd followed by cd - should be wrapped in a subshell "(...)" and the 'cd -' eliminated.

**Comment by zapashcanon** [comment]
Please read the title of the issue...

When using `cd -` there's already SC2013. The issue is about the fact that SC2013 is not raised when using nested cd.

**Comment by matthewpersico** [comment]
aaagghhh!!! I just spent 20 minutes trying to justify my argument when I realized the whole issue is WRONG: The id in question is 2103, not 2013. THAT's why this issue didn't make sense to me. Going to correct it if I can...

**Comment by koalaman** [comment]
This was because check had a list of code block typed it checked, and `(..)` subshell nodes had been accidentally left off. It was incidentally fixed in eeb7ea01c which rewrote the function to use the general purpose code block utility functions it should have been using all along. Thanks!
